### PR TITLE
Replace Travis with Github Actions

### PR DESCRIPTION
**Pull request by ktbarrett**
As discussed in the maintainers meeting. The runtime of the tests with Github Actions is 3x faster than with Travis. I have also increased the number of tests to cover a larger set of the matrix of OSes and simulators. And this lays the groundwork for even more tests.

I did not move the PyPI deployment over. That can be done in another PR, or in this one if I can get an API key for PyPI.

### CODE REVIEW

**Comment by cmarqu** [comment]
It's not easy to see the ``may_fail`` flag in the GHA output (AFAICS), maybe put it into the name until they improve that?

**Review by themperek — APPROVED** [review]
Looks OK to me.

File: .github/workflows/regression-tests.yml [hunk 1]
```diff
@@ -11,65 +11,278 @@ on:
...
+        python-version: ${{matrix.python-version}}
+    - name: flake8
+      run: |
+        pip install flake8
+        flake8
+
   tests:
-    name: ${{matrix.sim}} (${{matrix.sim-version}}) | ${{matrix.os}} | Python ${{matrix.python-version}}
+
+    name: ${{matrix.extra_name}}${{matrix.sim}} (${{matrix.sim-version}}) | ${{matrix.os}} | Python ${{matrix.python-version}} ${{matrix.may_fail || '| May Fail'}}
```
Comment (themperek) [inline .github/workflows/regression-tests.yml:45]: Does it show `true` when `may_fail` is false?

Comment (ktbarrett) [reply .github/workflows/regression-tests.yml:45]: You can see for yourself below. It does not. It shows 'May Fail' at the end or nothing.

Comment (themperek) [reply .github/workflows/regression-tests.yml:45]: I think have extra `true`.
"Regression Tests / icarus (master) | ubuntu-20.04 | Python 3.8 **true** (pull_request)" 

It does not matter.

Comment (ktbarrett) [reply .github/workflows/regression-tests.yml:45]: You are right, I totally got it backwards. I'll fix it.

**Comment by imphil** [comment]
Can you also resolve the warnings around add-path/set-path, e.g. 

"The `add-path` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/"

**Review by imphil — APPROVED** [review]
LGTM once you have it passing CI :-)

**Pull request merged**